src/vs/workbench/contrib/terminal/common/scripts/shellIntegration-rc.zsh

```diff
@@ -111,6 +115,22 @@ __vsc_update_cwd() {
 	builtin printf '\e]633;P;Cwd=%s\a' "$(__vsc_escape_value "${PWD}")"
 }
 
+__vsc_update_env() {
+	builtin printf '\e]633;EnvSingleStart;%s;\a' $__vsc_nonce
+	for var in ${(k)parameters}; do
+		if printenv "$var" >/dev/null 2>&1; then
+			value=$(builtin printf '%s' "${(P)var}")
+			# It is not valid to have hypen in variable name
+			# It is not valid to have hypen to start variable value
+			if [[ "$value" == -* || "$var" == *-*  ]]; then
+				continue
+			fi
+			builtin printf '\e]633;EnvSingleEntry;%s;%s;%s\a' "$var" "$(__vsc_escape_value "$value")" $__vsc_nonce
```

Comment:
> Going through more rigorous testing, I realized sometimes (although very not expected), things like -q show up as an environment variable's value. This of course throws error. I've aimed to better handle it now, soo it skips that variable entirely.

We want to make sure this works, it sounds like we're passing the string in as a space separated list of args, rather than a single string containing spaces. Try tweaking how $value is passed to that function.

> Also since hypens shouldn't really be part of variable name, we would skip the 'sending' of the variable if we detect usage of - in an environment variable name.

I don't think they're valid, but we also want to not special case this behavior either as well as it could hide obscure bugs.

Reply:
I agree, I also want to avoid putting special case in update_env function...
would something like [this](https://github.com/microsoft/vscode/pull/237977/commits/c919ff1f2cefaefe148acfaf4b657dd3602ac39c) be better? - Adding a elif in escape_method so that I would escape it in hex similar to already existing backslashes, semi-colons, newlines.

Reply:
I think trying to escape `-` wouldn't actually fix the underlying problem, which based on this:

![image](https://github.com/user-attachments/assets/24270d96-3c64-46e3-98d1-27cd1d51eb9f)

Appears to be that the `printf` call is getting a string as the first argument, and `-q` as the second one?

Does the problem still happen if we undo the special escape case? We don't want to add a case there unless it's really needed as it adds complexity and may hurt performance.